odl/util/testutils.py

```diff
@@ -433,13 +481,21 @@ def noise_elements(space, n=1):
     noise_array
     noise_element
     """
-    arrs = tuple(noise_array(space) for _ in range(n))
+    from odl.space.pspace import ProductSpace
+
+    if isinstance(space, ProductSpace) and not space.is_power_space:
+        raise ValueError('`space` cannot be a non-power product space')
```

Comment:
Why not? Part of the reason we use this function is that it supports these things (as compared to e.g. `space.element(np.random.rand)`)